### ISSUE TITLE
Xamarin not compatible

### ISSUE THREAD

**Issue by tedebus** [appccelerate/statemachine#5]
Is there any way (or prevision) to use Appccelerate on Xamarin SDK or Mono?
Thank you.

**Comment by ursenzler** [comment]
I don't know. I never worked with Xamarin or Mono.

The state machine is a PCL, but probably not with the right target platforms for Xamarin.

**Comment by ursenzler** [comment]
I switched the PCL to profile 78.
You can try the package from our staging environment: https://www.myget.org/F/appccelerate/api/v2/package/Appccelerate.StateMachine/2.13.0

If it works, I can release it to nuget.org.

**Comment by tedebus** [comment]
Hi! Thank you, but there is another issue: 
Original message (in italian):
Impossibile risolvere la dipendenza 'Appccelerate.Fundamentals (≥ 2.13 && < 3.0)'.
Translation should be:
Can't resolve dependency 'Appccelerate.Fundamentals (≥ 2.13 && < 3.0)'.

**Comment by tedebus** [comment]
Obviously I tried to install the dependency, but it can't be installed for the reasons above.

**Comment by ursenzler** [comment]
A sorry, you need the Appccelerate.Fundamentals package from the staging environment, too: https://www.myget.org/F/appccelerate/api/v2/package/Appccelerate.Fundamentals/2.13.0

**Comment by tedebus** [comment]
HI, thank you. I tried it but it seems not to work.
This is the result of Nuget installer:

Original version:
```
Installazione di 'Appccelerate.Fundamentals 2.13.0'.
Installazione di 'Appccelerate.Fundamentals 2.13.0' completata.
Aggiunta di 'Appccelerate.Fundamentals 2.13.0' a AppManagement.
Disinstallazione di 'Appccelerate.Fundamentals 2.13.0'.
Disinstallazione di 'Appccelerate.Fundamentals 2.13.0' completata.
Rollback in progress ...
Can not Install the package 'Appccelerate.Fundamentals 2.13.0'.
Si sta tentando di eseguire l'installazione in un progetto destinato a 'portable-net45+sl50+MonoAndroid10+xamarinios10+MonoTouch10',
ma il pacchetto non contiene riferimenti ad assembly o file di dati
compatibili con il framework.
Per ulteriori informazioni, contattare l'autore del pacchetto.
```
That in english should be:
```
Installing 'Appccelerate.Fundamentals 2.13.0'.
Installing 'Appccelerate.Fundamentals 2.13.0' completed.
Adding 'Appccelerate.Fundamentals 2.13.0' to AppManagement.
Uninstall 'Appccelerate.Fundamentals 2.13.0'.
Uninstall 'Appccelerate.Fundamentals 2.13.0' completed.
Installation failed. Rollback in progress ...
Can not install the package 'Appccelerate.Fundamentals 2.13.0'.
You are trying to install to a project intended to 'portable-net45+sl50+MonoAndroid10+xamarinios10+MonoTouch10',
but the package does not contain assembly references or data file
compatible with the framework.
For more information, contact the author of the package.
```
Thank you in advance

**Comment by ursenzler** [comment]
I think the problem is that your library targets also Silverlight 5. Fundamentals targes net45+win8+wp8.

If you really need sl5, I can try to support it, too.

**Comment by ursenzler** [comment]
Until we figure out the correct target, maybe you can copy the code of Appccelerate.Fundamentals and Appccelerate.StateMachine into your project so that you are not blocked by this issue.

**Comment by tedebus** [comment]
I don't need SL5, but the matter is that PCL destination manager includes it automatically. When I turn off it an info banner says: "Following elements will be automatically set as destination because they support the same portable API set: Silverlight 5"
Actually the destinations that I select are: MS .Net 4.5 framework, Xamarin.Android, Xamarin.iOS and Xamarin.iOS classic.
Even if I decide to unckeck .Net 4.5 Fw checkbox, it is automatically checked (as SL5).

**Comment by ursenzler** [comment]
PCL destination manager? Visual Studio or Xamarin Studio?

Do you mean this dialog?
![image](https://cloud.githubusercontent.com/assets/287565/7248935/f1d91616-e816-11e4-9824-e971613b03ff.png)

**Comment by tedebus** [comment]
Exactly this one.
I use VS2013
On your configuration iOS and Android destinations are missing.
Only last night I noted that is possible to select .Net (the first one) and Windows 8 without selecting SL, otherwise SL is automatically checked. With those options (.Net+Win8+Android+iOs) the packages are both installed from VS2013 but I didn't try to compile because the code was a little upside down for other reasons :)
As soon as I'll try I'll inform you.
Thank you so much!

**Comment by ursenzler** [comment]
You 're welcome.

It is strange that VS thinks that SL5 is compatible because supporting SL5 would be a bit of a nightmare due to a lot of missing APIs.

**Comment by ursenzler** [comment]
I asked a colleague with Xamarin to check it out. His dialog looks like:
![image](https://cloud.githubusercontent.com/assets/287565/7251307/74ebe3ae-e82a-11e4-929d-f29ce0341c91.png)

**Comment by tedebus** [comment]
That is (something is missing for me, i.e. XBox).
If you deselect WPhoneSL8 does it automatically select SL4?

**Comment by Gwanun** [comment]
Hello tedebus,

I'am the colleague of ursenzler having the Xamarin installation. I noticed I made a mistake in the previous screenshot: I started VS2012 instead of VS2013.

In VS2013, the target framework screen looks like this for me:

![image](https://cloud.githubusercontent.com/assets/5850963/7313630/4f4a3180-ea5c-11e4-9309-c5193e689856.png)

This configuration corresponds to PCL Profle 78. With that I was able to successfully import the new Appccelerate PCLs.

![image](https://cloud.githubusercontent.com/assets/5850963/7313721/192be246-ea5d-11e4-9ff6-5219fa9393c2.png)
